Projects/BricKuber/brickuber_lib.py

```diff
@@ -43,54 +56,73 @@ def __init__(self, robot_style, debug = False):
             self.MOTOR_GRAB_SPEED_GRAB = 400
             self.MOTOR_GRAB_SPEED_FLIP = 600
             self.MOTOR_GRAB_SPEED_REST = 400
+            
+            spin_direction = 1             
+            
         elif robot_style == "EV3":
             self.TurnTablePinion = 12
             self.TurnTableGear = 36
             
             # motor position constants (these likely need to be adjusted)
-            self.MOTOR_GRAB_POSITION_HOME      = 0
-            self.MOTOR_GRAB_POSITION_REST      = -35
-            self.MOTOR_GRAB_POSITION_FLIP_PUSH = -90
-            self.MOTOR_GRAB_POSITION_GRAB      = -130
-            self.MOTOR_GRAB_POSITION_FLIP      = -240
+            self.MOTOR_GRAB_POSITION_HOME      = -337 # -120 # 
+            self.MOTOR_GRAB_POSITION_REST      = -337 # -240 # 
+            self.MOTOR_GRAB_POSITION_FLIP_PUSH = -280 # -280 # 
+            self.MOTOR_GRAB_POSITION_GRAB      = -240 # -306 # 
+            self.MOTOR_GRAB_POSITION_FLIP      = -120 # -393 # 
             
             self.MOTOR_GRAB_SPEED_GRAB = 400
             self.MOTOR_GRAB_SPEED_FLIP = 600
             self.MOTOR_GRAB_SPEED_REST = 400
         else:
             raise ValueError("Unsupported robot style")
-        
+                
         self.BP.set_motor_limits(self.MOTOR_PORTS[self.MOTOR_TURN], 0, ((500 * self.TurnTableGear) / self.TurnTablePinion))
         
         self.home_all()
     
+    def read_encoder(self):
```

Comment:
Why have this method? It looks like it's only used for debugging, and by the name (and even printed text) it's not clear which encoder it's for.

Reply:
I built this for debugging.  I'll add a comment in there.  But for us mortals it helps get this setup correctly and figure out what's going on.